### PR TITLE
Fixed product test certification service URL

### PR DESCRIPTION
**Pull request by c-ryan-k**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

**Comment by c-ryan-k** [comment]
Integration tests for the certification service are a bit out of date (and were not updated due to inconsistencies between the test and production environments) - but an easy test for this fix is:

`az iot product test search --product-id [any_product_id]` will fail
`az iot product test search --product-id [product_id] --base-url https://prod.certsvc.trafficmanager.net` will succeed.

`az iot product requirement list` will fail
`az iot product requirement list --base-url https://prod.certsvc.trafficmanager.net` will succeed.

This is also the service request URL being used on `https://certify.azure.com`.

**Comment by vilit1** [comment]
> Integration tests for the certification service are a bit out of date (and were not enabled / maintained due to inconsistencies between the test and production environments) - but an easy test for this fix is:
> 
> `az iot product test search --product-id [any_product_id]` will fail `az iot product test search --product-id [product_id] --base-url https://prod.certsvc.trafficmanager.net` will succeed.
> 
> `az iot product requirement list` will fail `az iot product requirement list --base-url https://prod.certsvc.trafficmanager.net` will succeed.
> 
> This is also the service request URL being used on `https://certify.azure.com`.

Screenshots :D?

**Comment by c-ryan-k** [comment]
image

**Review by vilit1 — APPROVED** [review]

**Comment by digimaun** [comment]
> Integration tests for the certification service are a bit out of date (and were not enabled / maintained due to inconsistencies between the test and production environments) - but an easy test for this fix is:
> 
> `az iot product test search --product-id [any_product_id]` will fail `az iot product test search --product-id [product_id] --base-url https://prod.certsvc.trafficmanager.net` will succeed.
> 
> `az iot product requirement list` will fail `az iot product requirement list --base-url https://prod.certsvc.trafficmanager.net` will succeed.
> 
> This is also the service request URL being used on `https://certify.azure.com`.

The az iot product tests have been disabled for a very long time -- can we use a different product Id or test setup such that we can get the value of these tests (and have it integrated in CI)? Also we have this az iot product issue https://github.com/Azure/azure-iot-cli-extension/issues/486, wondering if there is a newer **public** API that can be used here.

**Comment by digimaun** [comment]
> > Integration tests for the certification service are a bit out of date (and were not enabled / maintained due to inconsistencies between the test and production environments) - but an easy test for this fix is:
> > `az iot product test search --product-id [any_product_id]` will fail `az iot product test search --product-id [product_id] --base-url https://prod.certsvc.trafficmanager.net` will succeed.
> > `az iot product requirement list` will fail `az iot product requirement list --base-url https://prod.certsvc.trafficmanager.net` will succeed.
> > This is also the service request URL being used on `https://certify.azure.com`.
> 
> The az iot product tests have been disabled for a very long time -- can we use a different product Id or test setup such that we can get the value of these tests (and have it integrated in CI)? Also we have this az iot product issue #486, wondering if there is a newer **public** API that can be used here.

A datapoint is https://certify.azure.com/ appears to be using the same version as in the CLI `2020-05-01-preview` 
image

**Comment by c-ryan-k** [comment]
> The az iot product tests have been disabled for a very long time -- can we use a different product Id or test setup such that we can get the value of these tests (and have it integrated in CI)? 

We can - though I did just run them again (everything except the all-up e2e test) and it worked for this product ID - probably just a timing issue in the existing test (or the last test run I made). Pulling them into CI is something we can do - provided we still have the ability to pass through a build if these fail intermittently - they're definitely flaky tests as of now.

> Also we have this az iot product issue #486, wondering if there is a newer **public** API that can be used here.

Like you mentioned below, the public certification site is still using the same API-version so I'm not sure there's an updated one but I will try to find out in the meantime.

File: azext_iot/product/shared.py [hunk 1]
```diff
@@ -44,4 +44,4 @@ class ValidationType(Enum):
     certification = "Certification"
 
 
-BASE_URL = "https://certify.azureiotsolutions.com"
+BASE_URL = "https://prod.certsvc.trafficmanager.net"
```
Comment (digimaun) [inline azext_iot/product/shared.py:47]: Is this URL publicly documented? Interesting there is no azure branding.

Comment (vilit1) [reply azext_iot/product/shared.py:47]: Traffic manager is an Azure product https://docs.microsoft.com/en-us/azure/traffic-manager/traffic-manager-overview

**Review by digimaun — APPROVED** [review]

**Review by digimaun — APPROVED** [review]
Approved with agreement to investigate windows agent issue.

**Pull request merged**